src/PSRule.Rules.Azure/rules/Azure.Redis.Rule.ps1

```diff
@@ -181,13 +181,28 @@ function global:IsEnterpriseCache {
     param ()
     process {
         return $Assert.In($TargetObject, 'sku.name', @(
-            'Enterprise_E10', 
-            'Enterprise_E20', 
-            'Enterprise_E50',
-            'Enterprise_E100',
-            'EnterpriseFlash_F300',
-            'EnterpriseFlash_F700',
-            'EnterpriseFlash_F1500'));
+                'Enterprise_E10', 
+                'Enterprise_E20', 
+                'Enterprise_E50',
+                'Enterprise_E100',
+                'EnterpriseFlash_F300',
+                'EnterpriseFlash_F700',
+                'EnterpriseFlash_F1500'));
+    }
+}
+
+function global:HasPublicNetworkAccess {
+    [CmdletBinding()]
+    param (
+        
+    )
+    process {
+        if ($PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled') {
+            $True
+        }
+        else {
+            $False
+        }
```

Comment:
Can be simplified to `return $PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled';` however we also need to check the case when the `Microsoft.Cache/redis/firewallRules` type is used.

If `Microsoft.Cache/redis/firewallRules` is used, we should just check it as normal.

```suggestion
        return $PSRule.TargetType -eq 'Microsoft.Cache/redis/firewallRules' -or ($PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled');
```